2.40.1_RN (#1167)
percona/pmm-doc: docs/release-notes/2.40.1.md

# Percona Monitoring and Management 2.40.1

| **Release date:** | Oct 18, 2023                                                                                    |
| ----------------- | ----------------------------------------------------------------------------------------------- |
| **Installation:** | [Installing Percona Monitoring and Management](https://www.percona.com/software/pmm/quickstart) |

Percona Monitoring and Management (PMM) is an open-source database monitoring, management, and observability solution for MySQL, PostgreSQL, and MongoDB.

## Fixed issues

- [PMM-12592](https://jira.percona.com/browse/PMM-12592) - Due to a known issue, the **Library panels** on the PMM dashboard were not working. The issue has been resolved now. 
- [PMM-12576](https://jira.percona.com/browse/PMM-12576) - After upgrading PMM from version 2.39.0 to 2.40.0 (not el7) using Docker, the admin user could not access the PMM UI. The issue has been resolved now.

> - [PMM-12587](https://jira.percona.com/browse/PMM-12587) - While trying to the upgrade from 2.39 to 2.40 it was breaking due to an issue with the grafana/database-migrator. The issue has been resolved now.

Reviewer: grafana/database-migrator is not a separate component of PMM; we should not mention it. 
Can we explain the "breaking" to the user better? What was happening in this case?

Author: @BupycHuk Could you please elaborate on this?

Reviewer 2: We had issues with having non correct mappings between dashboards and folders, users and groups and etc. so it was fixed.

> - [PMM-12587](https://jira.percona.com/browse/PMM-12587) - During the upgrade from version 2.39 to 2.40, the process was failing because of incorrect mappings between dashboards, folders, users, and groups. The issue has been resolved.

Reviewer 3: ```suggestion
- [PMM-12587](https://jira.percona.com/browse/PMM-12587) - During the upgrade from version 2.39.0 to 2.40.0, the process was failing because of incorrect mappings between dashboards, folders, users, and groups. The issue has been resolved.
```

# Percona Monitoring and Management 2.40.1

| **Release date:** | Oct 18, 2023                                                                                    |
| ----------------- | ----------------------------------------------------------------------------------------------- |
| **Installation:** | [Installing Percona Monitoring and Management](https://www.percona.com/software/pmm/quickstart) |

Percona Monitoring and Management (PMM) is an open-source database monitoring, management, and observability solution for MySQL, PostgreSQL, and MongoDB.

## Fixed issues

- [PMM-12592](https://jira.percona.com/browse/PMM-12592) - Fixed an issue where the **Library Panels** for the PMM dashboard were not working.

> - [PMM-12576](https://jira.percona.com/browse/PMM-12576) - After upgrading to PMM version 2.40.0, changing the Admin user's password from the terminal was not functioning. The issue has been resolved now. This impacts all versions prior to PMM 2.40.0.

Reviewer: "This impacts all versions prior to PMM 2.40.0." 
Is this?? 
I'm reading this as "in all versions before 2.40  user is not able to change the password. " and this is not the case, afaik

Reviewer 2: let's remove part about impact

Author: ok

> - [PMM-12587](https://jira.percona.com/browse/PMM-12587) - During the upgrade from version 2.39.0 to 2.40.0, the process failed with `500 internal server error` and an indefinite loop of `pmm-update-perform-init service` because of incorrect mappings between dashboards, folders, users, and groups. The issue has been resolved.

Reviewer: 1) indefinite loop of `pmm-update-perform-init service` - as s User I can't see this from UI, so this is redundant.
2) Was it always failing? As far as afaik - it can end up updating BUT!  the folders, users and their permissions are broken/mixed 

@BupycHuk PTAL

Reviewer 2: sometimes it end-up with indefinite loop. I think we shouldn't mention 2.39 here, because upgrade could be done from any version prior 2.40.0

Author: ok.

> - [PMM-12590](https://jira.percona.com/browse/PMM-12590) - Fixed CVE-2023-4911.

Reviewer 2: Should we explain internals and what we've done?

> - [PMM-12590](https://jira.percona.com/browse/PMM-12590) - Fixed CVE-2023-4911.

Reviewer: Well,  we need to be clear about what's here. 
I already have questions about the criticality of this item. And it's not really the case as. So we need some Informative but calm statement here. I see this as "better base image for PMM, protected  from this CVE"

# Percona Monitoring and Management 2.40.1

| **Release date:** | Oct 18, 2023                                                                                    |
| ----------------- | ----------------------------------------------------------------------------------------------- |
| **Installation:** | [Installing Percona Monitoring and Management](https://www.percona.com/software/pmm/quickstart) |

Percona Monitoring and Management (PMM) is an open-source database monitoring, management, and observability solution for MySQL, PostgreSQL, and MongoDB.

## Fixed issues

> - [PMM-12592](https://jira.percona.com/browse/PMM-12592) - Fixed an issue where the **Library Panels** for the PMM dashboard were not working.

Reviewer 2: after upgrade to 2.40.0

- [PMM-12576](https://jira.percona.com/browse/PMM-12576) - After upgrading to PMM version 2.40.0, changing the Admin user's password from the terminal was not functioning. The issue has been resolved now.

> - [PMM-12587](https://jira.percona.com/browse/PMM-12587) - During the upgrade to PMM 2.40.0, the process failed with `500 internal server error` and an indefinite loop of `pmm-update-perform-init service` because of incorrect mappings between dashboards, folders, users, and groups. The issue has been resolved.

Reviewer: ```suggestion
- [PMM-12587](https://jira.percona.com/browse/PMM-12587) - 
After the upgrade to 2.40.0  (can go successfully or with `500 internal server error` ), the User can face incorrect mappings between dashboards, folders, users, and groups compared to previous configurations. The issue has been resolved.
```

Reviewer: The main problem is broken mapping; all other information is secondary.

Author: fixed.

# Percona Monitoring and Management 2.40.1

> | **Release date:** | Oct 18, 2023                                                                                    |

Reviewer 2: ```suggestion
| **Release date:** | Oct 20, 2023                                                                                    |
```

| ----------------- | ----------------------------------------------------------------------------------------------- |
| **Installation:** | [Installing Percona Monitoring and Management](https://www.percona.com/software/pmm/quickstart) |

Percona Monitoring and Management (PMM) is an open-source database monitoring, management, and observability solution for MySQL, PostgreSQL, and MongoDB.

## Fixed issues

- [PMM-12592](https://jira.percona.com/browse/PMM-12592) - Fixed an issue where the **Library Panels** for the PMM dashboard were not working after upgrade to PMM 2.40.0.
- [PMM-12576](https://jira.percona.com/browse/PMM-12576) - After upgrading to PMM 2.40.0, changing the Admin user's password from the terminal was not functioning. The issue has been resolved now.
- [PMM-12587](https://jira.percona.com/browse/PMM-12587) - After upgrading to PMM 2.40.0, some users may experience incorrect mappings between dashboards, folders, users, and groups. This can result in either a successful upgrade or a `500 internal server error`. The issue has now been resolved.

> - [PMM-12590](https://jira.percona.com/browse/PMM-12590) - Fixed [CVE-2023-4911](https://nvd.nist.gov/vuln/detail/CVE-2023-4911).

Reviewer: ```suggestion
- [PMM-12590](https://jira.percona.com/browse/PMM-12590) [CVE-2023-4911](https://nvd.nist.gov/vuln/detail/CVE-2023-4911) is a vulnerability in the base OS that PMM is based on. It has been fixed in the base OS, and the fix is available in PMM.
```

## How to upgrade to PMM 2.40.1

For instructions on upgrading to PMM 2.40.1 and mitigating issues from PMM 2.40.0, follow the steps below:

1. Copy the file from `/srv/backup/grafana/grafana.db` to `/srv/grafana/grafana.db`

    ```sh
    docker exec -t pmm-server cp /srv/backup/grafana/grafana.db /srv/grafana/grafana.db
    ```
2. Set permissions:

    ```sh

>     chmod 640 /srv/grafana/grafana.db

Reviewer 2: these two commands are also should be called in docker
